File: filter-pattern/java/src/CriteriaFemale.java
package src;

import java.util.*;

// 根据标准接口实现的过滤
public class CriteriaFemale implements Criteria {

  @Override
  public List<Person> filter(List<Person> persons) {
    List<Person> femalePersons = new ArrayList<Person>();
    for (Person person : persons) {
      if (person.getGender().equalsIgnoreCase("FEMALE")) {
        femalePersons.add(person);
      }
    }
    return femalePersons;
  }
}

File: filter-pattern/ts/src/CriteriaFemale.js
export class CriteriaFemale {
    filter(persons) {
        const femalePersons = [];
        for (const person of persons) {
            if (person.getGender().toUpperCase() === 'FEMALE') {
                femalePersons.push(person);
            }
        }
        return femalePersons;
    }
}
